paclair/api/abstract_clair_requests.py

```diff
@@ -99,31 +99,8 @@ def get_ancestry_html(self, ancestry):
         :param ancestry: ancestry (name) to analyse
         :return: html
         """
-        clair_info = []
-        for feature in self._iter_features(self.get_ancestry_json(ancestry)):
-            for vuln in feature.get("vulnerabilities", {}):
-                vuln = InsensitiveCaseDict(vuln)
-                # metadata is a string in v3 and dict in v1
-                metadata = vuln.get("Metadata", {})
-                if isinstance(metadata, str):
-                    try:
-                        metadata = json.loads(metadata)
-                    except ValueError:
-                        metadata = {}
-                cvss = metadata.get('NVD', {}).get("CVSSv2", {})
-                cvss_vector = self.split_vectors(cvss.get('Vectors', ""))
-                clair_info.append({"ID": len(clair_info),
-                                   "CVE": vuln.get("Name"),
-                                   "SEVERITY": vuln.get("Severity"),
-                                   "PACKAGE": feature.get("Name"),
-                                   "CURRENT": feature.get("Version"),
-                                   "FIXED": vuln.get("FixedBy", ""),
-                                   "INTRODUCED": feature.get("AddedBy"),
-                                   "DESCRIPTION": vuln.get("Description"),
-                                   "LINK": vuln.get("Link"),
-                                   "VECTORS": cvss_vector,
-                                   "SCORE": cvss.get("Score")})
-        return template(self.html_template, info=clair_info)
+        raise NotImplementedError("Implement in sub classes")
```

Comment:
The only difference in the implementation between V1 and V3 is this line : 

V3: 
`"INTRODUCED": layer.get("layer").get("hash")`
V1:
`"INTRODUCED": feature.get("AddedBy"),`

Instead of duplicating code, _iter_features should return a tuple like that: (feature, layer_hash).
It will allow to implement a generic get_ancestry_html method in AbstractClairRequests.

Something like that:


    for feature, introduced_in in self._iter_features(self.get_ancestry_json(ancestry)):
        for vuln in feature.get("vulnerabilities", {}):
            ...
            "INTRODUCED": introduced_in,
            ...`